Dockerfile.cross

```diff
@@ -0,0 +1,297 @@
+FROM tini-base AS tini-base-bullseye
+RUN apt-get update && apt-get install -y lld
+ENV XX_CC_PREFER_LINKER=ld
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y binutils gcc dpkg-dev pkg-config
+
+FROM tini-base-${BASE_VARIANT} AS tini-build
+ARG TINI_VERSION
+RUN --mount=from=tini-src,src=/usr/src/tini,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$TINI_VERSION" \
+    && cmake -DCMAKE_C_COMPILER=$(xx-info)-gcc . \
+    && make tini-static \
+    && mkdir /out \
+    && mv tini-static /out/docker-init
+
+FROM binary-dummy AS tini-darwin
+FROM binary-dummy AS tini-freebsd
+FROM tini-build AS tini-linux
+FROM binary-dummy AS tini-windows
+FROM tini-${TARGETOS} AS tini
+
+# rootlesskit
+FROM build-base-${BASE_VARIANT} AS rootlesskit-src
+WORKDIR /usr/src
+RUN git clone https://github.com/rootless-containers/rootlesskit.git rootlesskit
+
+FROM build-base-${BASE_VARIANT} AS rootlesskit-base-static
+ENV CGO_ENABLED=0
+WORKDIR /go/src/github.com/rootless-containers/rootlesskit
+
+FROM build-base-${BASE_VARIANT} AS rootlesskit-base-dynamic
+ENV ROOTLESSKIT_LDFLAGS="-linkmode=external"
+WORKDIR /go/src/github.com/rootless-containers/rootlesskit
+
+FROM rootlesskit-base-${GO_LINKMODE} AS rootlesskit-base
+ARG TARGETPLATFORM
+RUN xx-go --wrap
+
+FROM rootlesskit-base AS rootlesskit-build
+ARG ROOTLESSKIT_VERSION
+ENV GOBIN=/out GO111MODULE=on
+COPY ./contrib/dockerd-rootless.sh /out/
+COPY ./contrib/dockerd-rootless-setuptool.sh /out/
+RUN --mount=from=rootlesskit-src,src=/usr/src/rootlesskit,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$ROOTLESSKIT_VERSION" \
+    && go build -o /out/rootlesskit -ldflags="$ROOTLESSKIT_LDFLAGS" -v ./cmd/rootlesskit \
+    && go build -o /out/rootlesskit-docker-proxy -ldflags="$ROOTLESSKIT_LDFLAGS" -v ./cmd/rootlesskit-docker-proxy
+
+FROM binary-dummy AS rootlesskit-darwin
+FROM binary-dummy AS rootlesskit-freebsd
+FROM rootlesskit-build AS rootlesskit-linux
+FROM binary-dummy AS rootlesskit-windows
+FROM rootlesskit-${TARGETOS} AS rootlesskit
+
+# containerutility
+FROM build-base-${BASE_VARIANT} AS containerutility-src
+WORKDIR /usr/src
+RUN git clone https://github.com/docker-archive/windows-container-utility.git containerutility
+
+FROM build-base-${BASE_VARIANT} AS containerutility-base-alpine
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc g++
+
+FROM build-base-${BASE_VARIANT} AS containerutility-base-bullseye
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y binutils gcc g++ dpkg-dev pkg-config
+
+FROM containerutility-base-${BASE_VARIANT} AS containerutility-build
+ARG CONTAINERUTILITY_VERSION
+RUN --mount=from=containerutility-src,src=/usr/src/containerutility,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$CONTAINERUTILITY_VERSION" \
+    && CC="$(xx-info)-gcc" CXX="$(xx-info)-g++" make \
+    && mkdir /out \
+    && mv containerutility.exe /out/
```

Comment:
Is this binary something that ships with `dockerd`?

Reply:
Oh was thinking it was bundled in https://download.docker.com/win/static/stable/x86_64/ but doesn't look like it. @thaJeztah what's the purpose of it?